src/rgw/rgw_putobj_processor.cc

```diff
@@ -200,14 +200,46 @@ int AtomicObjectProcessor::process_first_chunk(bufferlist&& data,
 
 int AtomicObjectProcessor::prepare()
 {
-  uint64_t max_chunk_size = 0;
-  int r = store->get_max_chunk_size(bucket_info.placement_rule, head_obj,
-                                    &max_chunk_size);
+  uint64_t max_head_chunk_size = 0;
+  uint64_t head_max_size;
+  uint64_t chunk_size = 0;
+  rgw_pool head_pool;
+
+  if (!store->get_obj_data_pool(bucket_info.placement_rule, head_obj, &head_pool)) {
+    return -EIO;
+  }
+
+  int r = store->get_max_chunk_size(head_pool, &max_head_chunk_size);
   if (r < 0) {
     return r;
   }
+
+  bool same_pool = true;
+
+  if (bucket_info.placement_rule != tail_placement_rule) {
+    rgw_pool tail_pool;
+    if (!store->get_obj_data_pool(tail_placement_rule, head_obj, &tail_pool)) {
+      return -EIO;
+    }
+
+    if (tail_pool != head_pool) {
+      same_pool = false;
+
+      r = store->get_max_chunk_size(tail_pool, &chunk_size);
+      if (r < 0) {
+        return r;
+      }
+
+      head_max_size = 0;
```

Comment:
hi, @yehudasa , is there any risk if we store data in head object? we are going to do a optimization: store head obj in 3-rep pool, store tail objs in EC pool.  see https://github.com/ceph/ceph/pull/16325

Reply:
There is no risk, however, it's not a clear cut whether you'd actually want that to happen. I started working on a change that would allow configuration of how much data would be stored in head object, but I want to get this in first due to time constraints.

Reply:
thank you for your help.